units/systemd-resolv-conf.service

```diff
@@ -0,0 +1,27 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Create resolv.conf if not already exists
+Documentation=man:systemd-resolved.service(8) man:resolv.conf(5)
+After=systemd-journald.service
+After=local-fs.target network-pre.target systemd-sysusers.service
+; no dynamic DNS configuration would be known before network configuration
+After=systemd-networkd.service
+Before=network.target
+ConditionPathExists=/run/systemd
+ConditionPathExists=!/etc/resolv.conf
+BindsTo=systemd-resolved.service
+
+[Service]
+ExecStart=-/bin/ln -s ../run/systemd/resolve/stub-resolv.conf resolv.conf
+RemainAfterExit=true
+ExecStop=/bin/sh -c "test $(realpath resolv.conf) == /run/systemd/resolve/stub-resolv.conf && rm -f resolv.conf"
```

Comment:
Since this is only executed if we successfully created the symlink (after "-" has been removed as described above), the check is not necessary. Just remove the file unconditionally.
(We don't want to call `realpath`: it's from coreutils, and would create a new dependency. Also, shell is not acceptable. But I don't think it's needed here at all.)

Reply:
Conditional removal were done to keep administrator done changes during systemd-resolved startup. If administrator changes target of /etc/resolv.conf to anything else, it must not remove his decision. Hence a bit cryptic removal. I would like to avoid administrator having to run ``systemctl stop systemd-resolv-conf`` before he manipulates /etc/resolv.conf. There are multiple services attempting to change it in addition to manual changes. This ensures it does not break resolution unintended. Though it makes it a bit ugly.

Reply:
The comment about `sh` and `realpath` still stands. Adding a shell script and a dependency on coreutils is not acceptable. This must be implemented in C. As mentioned before, it would be easiest to just implement a mode in `systemd-resolved` binary itself (or maybe `resolvectl`?) that creates the symlink.